### PR TITLE
Allow to easily pip-install tool

### PR DESCRIPTION
**Pull request by ttylec**
I've added setup.py and slightly altered file structure. Now you can simply:
```
$ pip install -U . 
```

from within repo, or from remote using git+ssh/git+https protocol.

The version is set using setuptools_scm, i.e. it will be either git tag or git tag+devSHA when not on tag. More or less.

### CODE REVIEW

**Comment by LuisBL** [comment]
Hello @davidfraser do you think you can accept the path ?

I would like to have it in with the standard 

    $ pip install pyan

**Comment by lainiwa** [comment]
In case someone will end here searching for pip install'able solution:
```sh
pip install git+https://github.com/ttylec/pyan
```
until this PR approved (as @davidfraser seems to be inactive)

**Comment by MalloZup** [comment]
It could be usefull to add a link to Readme.doc and close this issue then

**Comment by ttylec** [comment]
But it will not resolve the issue... depending on my fork is not a long term solution.

**Comment by MalloZup** [comment]
A right @ttylec  I have overlooked the fact that the link proposed is not the same here

File: scripts/pyan [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+from glob import glob
+from optparse import OptionParser  # TODO: migrate to argparse
+
+from pyan.analyzer import CallGraphVisitor
+from pyan.visgraph import VisualGraph
+from pyan.writers import TgfWriter, DotWriter, YedWriter
+
+
+def main():
+    usage = """usage: %prog FILENAME... [--dot|--tgf|--yed]"""
```
[the remark this answers is not in the document]
Comment (davidfraser) [reply scripts/pyan]: (Sorry I'd made a comment but didn't realize that it didn't show up until you completed the review)

[the remark this answers is not in the document]
Comment (ttylec) [reply scripts/pyan]: That's a very good question :) It was long time ago... I either forgot to remove main.py or decided to be least invasive in project structure and just add parts that allow easy pip-install. Anyway, I pushed change that removes the `pyan/main.py` file as obsolete.

**Comment by ttylec** [comment]
Applied requested change: main.py (and code duplication) removed.

**Comment by Technologicat** [comment]
Hi, Pyan3 current maintainer here. Technically speaking, @davidfraser maintains this main repo, but the Python 3 conversion and most of the recent code in Pyan is from me.

I just noticed this. We should perhaps pool our efforts, as I've already made [my development fork](https://github.com/Technologicat/pyan) pip-installable (and pushed it to PyPI as [pyan3](https://pypi.org/project/pyan3/)) a while back, after it was requested in https://github.com/Technologicat/pyan/issues/6. At that time it didn't occur to me there was an issue about it in the main repo, too.

I don't have a strong opinion which install script it should be, as long as we **have** one. If yours is better, I'll happily switch over.

My development fork is every now and then manually synced to @davidfraser's main repo via a PR, but as of this writing I have some semi-recent developments that I haven't gotten around to packaging yet. Maybe the optimal time to do that is after we resolve this.

My main concern is, Pyan should be installable from PyPI without providing any extra command-line parameters to `pip`, like serious Python packages generally are. So we need to have just one official package on PyPI, where the homepage metadata points to the official GitHub fork. It should probably be @davidfraser's.

For the near future, I can sync the changes manually every now and then, and push an updated `pyan3` package to PyPI, but for the long term, we need a better solution. Ideas?

**Comment by ttylec** [comment]
"My" script is just David's main.py moved to script + setup.py to install it. Since you made more modifications, let's stick with that (I was not aware of yours fork either! :).

The svg is small addition, I can PR just that to yours repo.

The version in setup.py -- you're right that using setuptools_scm is terse, but works nicely only from within the git repo. Honestly, I don't know what's the best method to handle version on PyPI. Maybe the most obvious one, i.e. manual entry on tagged releases?

BTW, I found some issue with your fork when I tried to pip install it from pypi. https://github.com/Technologicat/pyan/issues/16

**Comment by Technologicat** [comment]
Yes, I noticed `pyan.py` got moved to `pyan3`. :)

My fork is a bit difficult to find mainly for historical reasons.

I did [some work on the original Pyan back in 2012](https://ejrh.wordpress.com/2012/08/18/coloured-call-graphs/), after which my development work took me into a direction where I didn't need the tool for some years. (Basically greenfielding some numerical solvers, so no need to analyze existing code.)

At that point, Pyan was still pretty much an experiment. Both EJRH (Edmund Horner, the original author of the first Pyan) and me had our Pyan codes tangled with other, completely unrelated projects in personal repos labeled, pretty much, as "miscellaneous". The code wasn't yet on GitHub.

This is when David stepped in. He extracted the Pyan files from our repos, and set up this official GitHub repo for Pyan. He also performed some maintenance and started curating PRs. Pyan started to be noticed and seen as an at least somewhat serious tool.

The commit history says it was November 2017, when Pyan became relevant for my work again. [The clock](https://pythonclock.org/) was ticking on Python 2, and (if I recall correctly) I'd done all my development in Python 3 already for a year or so at that point.

But in 2017, Pyan didn't yet support Python 3. Goodbye `compiler` ([and good riddance!](https://stackoverflow.com/questions/909092/why-is-the-compiler-package-discontinued-in-python-3)), hello `ast`! The new API was rather different, so the analyzer needed an overhaul. [Green Tree Snakes](https://greentreesnakes.readthedocs.io/en/latest/nodes.html), the missing AST documentation, came in highly useful.

Whence Pyan3. I noticed David had set up a repository, so I did the conversion work - to get Pyan working for my own use cases again - and PR'd the changes to him.

In practice what has happened since then is that David curates this official repo, while new development for Pyan3 occurs in my development fork. Neither of us is very active - I have many other projects (both open source and work), and I'm sure the same goes for him.

I send PRs when I get around to it - David has said to keep them coming, he'll merge my changes - but since it's a manual process, there may occasionally be long delays between syncs to the main repo.

So, that's the history of Pyan in a nutshell :)

**Comment by Technologicat** [comment]
As for managing the version number, in all the projects I have made PyPI packages for, I've tagged the revision manually in git once I'm sure `mypackage/__init__.py` has the correct `__version__` in it (in practice, in my own projects I tend to bump `__version__` as the first commit after each release, to make sure I won't forget). The rest can be automated.

To make the release, a human would pick up the correct version number by opening `mypackage/__init__.py` in an editor, then at its top level look for an assignment statement that writes to the name `__version__`, and then pick the string that appears on the right-hand side. That translates to ~10 lines of Python. :)

This makes sure that the version number seen by PyPI is in sync with that in `mypackage/__init__.py`, so you can `python3 -c 'import mypackage; print(mypackage.__version__)'` and it's guaranteed to be correct.

(Whether that's the best approach, is indeed a different question.)

**Pull request closed without merging**